scripts/dts/gen_defines.py

```diff
@@ -599,6 +603,44 @@ def write_clocks(node):
                 controller.props["clock-frequency"].val)
 
 
+def write_dmas(node):
+    if "dmas" not in node.props:
+        return
+
+    write_phandle_val_list(node.props["dmas"])
+
+    # Client parameter syntax may vary with the DMA controller changes.
+    for dma_i, dma in enumerate(node.props["dmas"].val):
+        if "st,stm32-dma" in dma.controller.compats:
+            dma.slot = dma.data["slot"]
+            dma.channel_config = dma.data["channel_config"]
+            dma.features = dma.data["features"]
```

Comment:
Could use plain local variables (`slot`, `channel_config`, etc.) The edtlib objects should probably treated as immutable.

Reply:
Would be nice to check that these keys exist ("channel_config", etc.) too, and exit with an error message otherwise. Check how the `err()` function is used.

The goal is for the new scripts to always fail with an error message, and never with a generic Python error.